scene/main/window.cpp

```diff
@@ -833,49 +859,34 @@ bool Window::is_visible() const {
 	return visible;
 }
 
-void Window::_update_window_size() {
-	// Force window to respect size limitations of rendering server.
```

Comment:
This PR indirectly limits the size:

1. Make `min_size` and `max_size` follow the size limit of the rendering server, see `Window::_clamp_limit_size`;
2. Then make sure that `max_size_used` is not less than `min_size`, see `Window::_validate_limit_size()`.

`max_size_used` is the upper limit of actual use, it will be clamped by `min_size` and `max_size`, combined with 1, it will eventually follow the size limit of the rendering server.

`get_clamped_minimum_size()` returns the actual lower bound used, which takes into account the case when `wrap_controls` is enabled.

Reply:
>     2. Then make sure that `max_size_used` is not less than `min_size`, see `Window::_validate_limit_size()`.

It just ensures that when `max_size` is not as large as `min_size`, `RS::get_singleton()->get_maximum_viewport_size()` is actually used. I'm not sure whether this method returns `(0,0)`.

Reply:
Since this is not an obvious functionality, I believe, that it should be documented in a comment.

Reply:
OK, some comments have been added.